modules/dnn/include/opencv2/dnn/csl/cublas.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_DNN_CSL_CUBLAS_HPP
+#define OPENCV_DNN_CSL_CUBLAS_HPP
```

Comment:
Do we really need these wrappers in a public OpenCV API under `modules/dnn/include`?

Private headers can be safely placed under "src" directory.

Reply:
I was attempting to allow users to write CUDA implementations for custom layers which requires that the forward and init methods to be added to the `Layer` class. These methods require some components from the CSL which I exposed publicly. I would have to expose everything to aid custom layers to be supported. I postponed the custom layer work to the end of the project. Hence, only the components that are required for `initCUDA` and `forwardCUDA` methods are in the public API directory.